### ISSUE TITLE
assertion failure: management.azure.com != api-dogfood.resources.windows-int.net

### ISSUE THREAD

**Issue by ctaggart** [Azure/azure-cli-dev-tools#210]
I'm recording tests against another a test Azure Cloud instance. When I try to playback the tests, I get host assertion failures like:

```
E           1 - (<Request (GET) https://api-dogfood.resources.windows-int.net/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/cli_test_vmware000001/providers/Microsoft.AVS/privateClouds/cloud1?api-version=2019-08-09-preview>).
E           Matchers succeeded : ['method', 'scheme', 'port', 'path', '_custom_request_query_matcher']
E           Matchers failed :
E           host - assertion failure :
E           management.azure.com != api-dogfood.resources.windows-int.net
```

See [them here](https://dev.azure.com/azure/az-vmware-cli/_build/results?buildId=10779&view=logs&j=19c563d8-3dcd-57d1-cd5e-9dd946b0a29b&t=6940a1e1-ee7a-5a07-3a75-fd4d55f85437). They are from this pull request: https://github.com/Azure/az-vmware-cli/pull/42 .

How do I disable the host assertion?

**Comment by haroldrandom** [comment]
Have you ran tests in live mode and playback again?

**Comment by ctaggart** [comment]
@haroldrandom Yes, I ran it in live mode and the tests pass after about 10 minutes with `azdev test VmwareScenarioTest --live`.  I then run `azdev test VmwareScenarioTest` and I get that error locally as well in CI for https://github.com/Azure/az-vmware-cli/pull/31.

**Comment by ctaggart** [comment]
I dug into this a bit. The [scenario base tests](https://github.com/Azure/azure-python-devtools/blob/master/doc/scenario_base_tests.md) use [vcr](https://github.com/vcr/vcr). There is a [host matcher](https://vcrpy.readthedocs.io/en/latest/api.html#vcr.matchers.host), but I'm not sure how to disabled it. The configuration of VCR looks to be done [in base.py here](https://github.com/Azure/azure-python-devtools/blob/3d9e0d945f8f2b021e1000ba17750b1adbf2884e/src/azure_devtools/scenario_tests/base.py#L106-L114).

**Comment by ctaggart** [comment]
I made this workaround:
``` python
class VmwareScenarioTest(ScenarioTest):
    def setUp(self):
        # https://vcrpy.readthedocs.io/en/latest/configuration.html#request-matching
        self.vcr.match_on=['scheme', 'method', 'path', 'query'] # not 'host', 'port'
        super(VmwareScenarioTest, self).setUp()
```
